dynamic_programming/knapsack.py

```diff
@@ -27,7 +27,7 @@ def mf_knapsack(i, wt, val, j):
 
 
 def knapsack(w, wt, val, n):
-    dp = [[0 for i in range(w + 1)] for j in range(n + 1)]
+    dp = [[0 for _ in range(w + 1)] for _ in range(n + 1)]
```

Comment:
Is some tool recommending that we do this?!?  If not then let's not do it as this **_reuse_** is unsettling.
```suggestion
    dp = [[0] * (w + 1) for _ in range(n + 1)]
```

Reply:
You're right. This seems more reasonable. Thanks.